staging/src/k8s.io/apimachinery/pkg/api/validate/safetraverse/safetraverse.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package safetraverse
+
+// Value takes a pointer to any value, which may or may not be nil,
+// and a function that traverses to another value (a typical use case is to dereference a non-nilable field),
+// and returns the result of the traversal, or returns nil if the given value is nil.
+// This is roughly equivalent to "value ? &fn(value) : nil" in languages that support the ternary operator.
+func Value[V any, T any](value *V, fn func(V) T) *T {
+	if value == nil {
+		return nil
+	}
+	o := fn(*value)
```

Comment:
Why deref?  I'd just pass it as a pointer.

Reply:
Good call.  I've simplified this down to only requiring a single `safe.Field()` function.